### ISSUE TITLE
Low Accuracy in workout in smaller area

### ISSUE THREAD

**Issue by Chidhambararajan** [OpenTracksApp/OpenTracks#534]
Hi , first I would like to say that the app is awesome
but when I do my workout in a confined area , there is some accuracy issues

So to test , I started tracking my workout in both Samsung Health and OpenTracks in parallel for comparison
(I have observed Samsung health to be accurate)

Results of workout in Small Area :

OpenTracks : 
    distance : 618m
    
Samsung Health :
    distance : 2.9km

Results of workout in Large Area :

OpenTracks :
    distance : 1.6km
    
Samsung Health : 
    distance : 1.4km

As you can see , the accuracy is good in large area workouts , but bad in small area workouts

I think that this difference is caused due to OpenTracks capturing location information in a lesser frequency than Samsung Health

Is there a way to fix it?

By the way I really love the features the app provides in a privacy friendly way

**Comment by dennisguse** [comment]
Not too much we can do (easily) [except setting GPS frequency to 1s and that will not increase the accuracy].

TL;DR
OpenTracks only uses the GPS.
The GPS measurements can be improved by using accelerometer, gyroscope, and barometric sensor.
But the sensor fusion is _really_ tricky.
Sadly, the Google PlayServices provide these, but we cannot use them.
And I guess Samsung Health either uses PlayServices or has some fancy algorithm in place.

If you know somebody doing a PhD or something similar in this domain tell them to help us.
This problem is a mixture of math, computer science, and understanding how the sensors works (incl. there limitations).

And sensor fusion is tricky...
For elevation gain estimation, we just use the barometric sensor as integrating a relative sensor (barometer with little noise, but a lot of drift) with an absolute sensor (huge noise, no drift) is already (too) complicated.

PS: Thanks ><
